src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/http_client.html.twig

```diff
@@ -121,11 +124,28 @@
                                     </tr>
                                     </tbody>
                                 </table>
+                                {# tmp: introspect the "trace" #}
+                                {{ dump(trace) }}
                             {% endfor %}
                         {% endif %}
                     </div>
                 </div>
             {% endfor %}
         {% endif %}
     </div>
+
+    <script type="text/javascript">//<![CDATA[
+
+        document.querySelectorAll('[data-clipboard-text]').forEach(function(button) {
+            button.addEventListener('click', function() {
+                navigator.clipboard.writeText(button.getAttribute('data-clipboard-text'));
+            })
+        });
+
+    //]]></script>
+
 {% endblock %}
+
+{% macro render_curl(trace) %}
+curl "{{ trace.url }}" --compressed -H "Connection: keep-alive" -v
```

Comment:
if we can provide a reusable service (request to curl) that be neat :) i believe there are some CurlBuilders in the wild to take inspiration from.

Reply:
well, I would say that it must at least include the method and the headers. For the body, that's harder.

Also, be careful about the fact that you need to escape `"` in the command arguments (or even better apply proper shell escaping, which might imply moving this logic to a helper method in the collector class rather than doing it in a twig macro)

Reply:
> moving this logic to a helper method in the collector class 

would there be any reason to not make it a first class feature, if we want this feature for the profiler?

a usecase i have in mind is adding the curl command in some log entries its context. Or perhaps display in CLI verbose mode.

Reply:
well, if you want to make it a first-class feature, the argument of this API cannot be the internal data structure of this collector

Reply:
Maybe it will be easier in the datacollector, agreed i'll move this

Reply:
For a ref, the curl formatter used by httplug: https://github.com/php-http/message/blob/master/src/Formatter/CurlCommandFormatter.php

Reply:
i'll check this thx